java.lang.IndexOutOfBoundsException: Index: 0, Size: 0    at openfoodfacts.github.scrachx.openfood.views.ContinuousScanActivity$2.onSuccess(ContinuousScanActivity.java:303)

```
2019-02-10 17:52:51.112 10825-10825/openfoodfacts.github.scrachx.openfood E/AndroidRuntime: FATAL EXCEPTION: main
    Process: openfoodfacts.github.scrachx.openfood, PID: 10825
    java.lang.IllegalStateException: Fatal Exception thrown on Scheduler.
        at io.reactivex.android.schedulers.HandlerScheduler$ScheduledRunnable.run(HandlerScheduler.java:111)
        at android.os.Handler.handleCallback(Handler.java:873)
        at android.os.Handler.dispatchMessage(Handler.java:99)
        at android.os.Looper.loop(Looper.java:193)
        at android.app.ActivityThread.main(ActivityThread.java:6718)
        at java.lang.reflect.Method.invoke(Native Method)
        at com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run(RuntimeInit.java:493)
        at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:858)
     Caused by: java.lang.IndexOutOfBoundsException: Index: 0, Size: 0
        at java.util.ArrayList.get(ArrayList.java:437)
        at openfoodfacts.github.scrachx.openfood.views.ContinuousScanActivity$2.onSuccess(ContinuousScanActivity.java:303)
        at openfoodfacts.github.scrachx.openfood.views.ContinuousScanActivity$2.onSuccess(ContinuousScanActivity.java:183)
        at io.reactivex.internal.operators.single.SingleDoOnSubscribe$DoOnSubscribeSingleObserver.onSuccess(SingleDoOnSubscribe.java:77)
        at io.reactivex.internal.operators.single.SingleObserveOn$ObserveOnSingleObserver.run(SingleObserveOn.java:81)
        at io.reactivex.android.schedulers.HandlerScheduler$ScheduledRunnable.run(HandlerScheduler.java:109)
        at android.os.Handler.handleCallback(Handler.java:873) 
        at android.os.Handler.dispatchMessage(Handler.java:99) 
        at android.os.Looper.loop(Looper.java:193) 
        at android.app.ActivityThread.main(ActivityThread.java:6718) 
        at java.lang.reflect.Method.invoke(Native Method) 
        at com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run(RuntimeInit.java:493) 
        at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:858) 
```

--- a/app/src/main/java/openfoodfacts/github/scrachx/openfood/views/ContinuousScanActivity.java
+++ b/app/src/main/java/openfoodfacts/github/scrachx/openfood/views/ContinuousScanActivity.java
@@ -290,16 +290,18 @@ public class ContinuousScanActivity extends android.support.v7.app.AppCompatActi
                             }
                             if(product.getEnvironmentImpactLevelTags()!=null) {
                                 List<String> tags=product.getEnvironmentImpactLevelTags();
-                                String tag=tags.get(0).replace("\"","");
-                                co2Icon.setVisibility(View.VISIBLE);
-                                if(tag.equals("en-high")){
-                                    co2Icon.setImageResource(R.drawable.ic_co2_high_24dp);
-                                } else if(tag.equals("en-low")){
-                                    co2Icon.setImageResource(R.drawable.ic_co2_low_24dp);
-                                } else if(tag.equals("en-medium")){
-                                    co2Icon.setImageResource(R.drawable.ic_co2_medium_24dp);
-                                } else {
-                                    co2Icon.setVisibility(View.GONE);
+                                if(tags.size() > 0) {
+                                    String tag=tags.get(0).replace("\"","");
+                                    co2Icon.setVisibility(View.VISIBLE);
+                                    if(tag.equals("en-high")){
+                                        co2Icon.setImageResource(R.drawable.ic_co2_high_24dp);
+                                    } else if(tag.equals("en-low")){
+                                        co2Icon.setImageResource(R.drawable.ic_co2_low_24dp);
+                                    } else if(tag.equals("en-medium")){
+                                        co2Icon.setImageResource(R.drawable.ic_co2_medium_24dp);
+                                    } else {
+                                        co2Icon.setVisibility(View.GONE);
+                                    }
                                 }
                             }
                             FragmentManager fm = getSupportFragmentManager();

--- a/app/src/main/java/openfoodfacts/github/scrachx/openfood/views/product/summary/SummaryProductFragment.java
+++ b/app/src/main/java/openfoodfacts/github/scrachx/openfood/views/product/summary/SummaryProductFragment.java
@@ -652,16 +652,18 @@ public class SummaryProductFragment extends BaseFragment implements CustomTabAct
             }
             if(product.getEnvironmentImpactLevelTags()!=null) {
                 List<String> tags=product.getEnvironmentImpactLevelTags();
-                String tag=tags.get(0).replace("\"","");
-                co2Icon.setVisibility(View.VISIBLE);
-                if(tag.equals("en-high")){
-                    co2Icon.setImageResource(R.drawable.ic_co2_high_24dp);
-                } else if(tag.equals("en-low")){
-                    co2Icon.setImageResource(R.drawable.ic_co2_low_24dp);
-                } else if(tag.equals("en-medium")){
-                    co2Icon.setImageResource(R.drawable.ic_co2_medium_24dp);
-                } else {
-                    co2Icon.setVisibility(View.GONE);
+                if(tags.size()>0) {
+                    String tag=tags.get(0).replace("\"","");
+                    co2Icon.setVisibility(View.VISIBLE);
+                    if(tag.equals("en-high")){
+                        co2Icon.setImageResource(R.drawable.ic_co2_high_24dp);
+                    } else if(tag.equals("en-low")){
+                        co2Icon.setImageResource(R.drawable.ic_co2_low_24dp);
+                    } else if(tag.equals("en-medium")){
+                        co2Icon.setImageResource(R.drawable.ic_co2_medium_24dp);
+                    } else {
+                        co2Icon.setVisibility(View.GONE);
+                    }
                 }
             }
 

